### PR TITLE
Anti-CSRF Handling should always account for partial matching

### PR DESCRIPTION
**Pull request by kingthorin**
- ExtensionAntiCSRF - Ensure public method always accounts for partial matching.
- ExtensionAntiCSRFUnitTest - Add a method to assert the behavior.

Related to:
- zaproxy/zap-extensions#5196
- zaproxy/zaproxy#8280

### CODE REVIEW

File: zap/src/test/java/org/zaproxy/zap/extension/anticsrf/ExtensionAntiCSRFUnitTest.java [hunk 1]
```diff
@@ -178,6 +178,30 @@ void shouldGetTokensFromInputFieldsIfIdIsAKnownToken() {
             assertAntiCsrfToken(tokens.get(2), message, KNOWN_TOKEN_3, "value3", 1);
         }
 
+        @Test
+        void shouldGetTokensFromInputFieldsIfKnownTokenWithPartialMatch() {
```
Comment (thc202) [inline zap/src/test/java/org/zaproxy/zap/extension/anticsrf/ExtensionAntiCSRFUnitTest.java:182]: This is already covered by `shouldGetTokensFromInputFieldsIfIdAndNameIncludesAKnownTokenAndPartialMatching*`. Though still worth adding one for `isAntiCsrfToken(String)` specifically.

Comment (kingthorin) [reply zap/src/test/java/org/zaproxy/zap/extension/anticsrf/ExtensionAntiCSRFUnitTest.java:182]: On it, will drop the println too :grinning:

**Comment by kingthorin** [comment]
Also I noticed one of the constants in the test class is misspelled, which I'm happy to correct here or another PR.

`UNKOWN_TOKEN` ➡`UNKNOWN_TOKEN` (missing N in known).

**Comment by thc202** [comment]
Fine here in a separate commit?

**Comment by kingthorin** [comment]
Sure

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**